core/src/main/java/org/apache/calcite/tools/RelRunners.java

```diff
@@ -29,6 +35,19 @@ private RelRunners() {}
 
   /** Runs a relational expression by creating a JDBC connection. */
   public static PreparedStatement run(RelNode rel) {
+    final RelShuttle shuttle = new RelHomogeneousShuttle() {
+      @Override public RelNode visit(TableScan scan) {
+        final RelOptTable table = scan.getTable();
+        if (scan instanceof LogicalTableScan
+            && Bindables.BindableTableScan.canHandle(table)) {
+          // Always replace the LogicalTableScan with BindableTableScan
+          // because it's implementation does not require a "schema" as context.
+          return Bindables.BindableTableScan.create(scan.getCluster(), table);
```

Comment:
This looks odd.
Can you clarify why is this replacement needed?

Reply:
Because the Enumerables implementation needs a "schema" to look up for the table. but the RelRunners does not actually have a schema, the Bindables actually can implement that.

Reply:
Can the clients of Relrunners do the appropriate conversion instead?

org.apache.calcite.tools.RelRunners#run API says nothing on what changes would it make to the input rel node.
It might be very unexpected for those who see just the call to `RelRunners.run(root))`.

Reply:
The logic is the same compared to the old one. The RelRunners always run a logical plan.